### PR TITLE
Introduce a link between execution and file as a dedicated table

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- added table `lnk_execution_file`
- migrated `Execution.additonalFiles` to using link table

It's part of https://github.com/saveourtool/save-cloud/pull/1670

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/FileInternalController.kt [hunk 1]
```diff
@@ -24,42 +26,21 @@ import reactor.kotlin.core.publisher.toMono
 @RestController
 @RequestMapping("/internal/files")
 class FileInternalController(
-    private val fileStorage: MigrationFileStorage,
+    private val fileStorage: NewFileStorage,
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/FileInternalController.kt]: need to rollback, we will use the old endpoints in phase 1

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt [hunk 2]
```diff
@@ -1,54 +1,60 @@
...
+import reactor.core.publisher.Mono
+import java.nio.ByteBuffer
+import java.time.Instant
+import javax.annotation.PostConstruct
 
 /**
  * Storage for evaluated tools are loaded by users
  */
 @Service
 class FileStorage(
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt]: need to rollback

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/OldFileStorage.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package com.saveourtool.save.backend.storage
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/OldFileStorage.kt]: should be removed

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Execution.kt [hunk 4]
```diff
@@ -85,7 +85,7 @@ class Execution(
 
     var sdk: String,
 
-    var additionalFiles: String,
+    var additionalFiles: String = "",
```
Comment (nulls) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Execution.kt]: can be removed

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/FileInternalController.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/FileInternalController.kt]: done

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/OldFileStorage.kt (hunk 3 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/OldFileStorage.kt]: done

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt [hunk 5]
```diff
@@ -287,6 +289,26 @@ class ExecutionService(
         return savedExecution
     }
 
+    /**
+     * Unlink provided [File] from all [Execution]s
+     *
+     * @param file
+     */
+    @Transactional
+    fun unlinkFileFromAllExecution(file: File) {
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt:298]: It's not actually an `unlink`, isn't it? We don't remove corresponding records from `lnkExecutionFileRepository`
So, maybe call it like `markeExecutionsCorrespondingToFileAsObsolete`?

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt (hunk 5 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt:298]: actually, it should delete `lnkExecutionFileRepository`.
Fixed the method

**Pull request merged**